.github/workflows/main.yaml

```diff
@@ -4,7 +4,10 @@
 
 name: Build and test linkml-model
 
-on: [pull_request]
+on:
+  pull_request:
+    branches: [ main ]
+    types: [ opened, synchronize, reopened ]
```

Comment:
I think it's fine to explicitly specify the activity types here, but just FYI these are the default anyway:

> By default, a workflow only runs when a `pull_request` event's activity type is `opened`, `synchronize`, or `reopened`. To trigger workflows by different activity types, use the types keyword. 

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request